packages/runtime/src/widgets/TabBar.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { WidgetRegistry } from "../registry";
+import { EnsembleRuntime } from "../runtime";
+import { Tabs, ConfigProvider } from "antd";
+import { Widget } from "framework";
+import { IconProps } from "../util/types";
+import { Icon } from "./Icon";
+const { TabPane } = Tabs;
+
+export type items = {
+  label: string;
+  icon?: IconProps;
+  widget: Widget;
+};
+export type TabBarProps = {
+  id?: string;
+  selectedIndex?: string;
+  items: items[];
+  styles?: {
+    tabPosition: "start" | "stretch";
+    tabPadding: string;
+    tabFontSize: number;
+    tabFontWeight: string;
+    tabBackgroundColor: string;
+    activeTabColor: string;
+    dividerColor: string;
+    inactiveTabColor: string;
+    indicatorColor: string;
+    indicatorThickness: string;
+  };
+};
+export const TabBar: React.FC<TabBarProps> = (props) => {
+  const renderLabel = (label: string, icon?: IconProps) => {
+    return (
+      <div
+        style={{
+          display: "flex",
+          alignItems: "center",
+          padding: props.styles?.tabPadding ?? "5px 5px 5px 5px",
+        }}
+      >
+        {icon && <Icon name={icon.name} size={icon.size} color={icon.color} />}{" "}
+        &nbsp; {label}
+      </div>
+    );
+  };
+
+  const customStyles = `
+    .ant-tabs-ink-bar {
+      height: 5px;
+      background: transparent;
+    }
+
+    .ant-tabs-ink-bar::after {
+      content: " ";
+      position: absolute;
+      left: 50%;
+      right: 0;
+      bottom: 0;
+      height: ${props.styles?.indicatorThickness || "3px"};
+      background: ${props.styles?.indicatorColor || "black"};
+      width: 100%;
+      transform: translateX(-50%);
+    }
+
+    .ant-tabs-top > .ant-tabs-nav::before {
+      border-bottom: 1px solid ${props.styles?.dividerColor || "grey"};
+    }
+
+    .ant-tabs > .ant-tabs-nav {
+      background-color: ${props.styles?.tabBackgroundColor || "white"};
+    }
+
+    .ant-tabs {
+      font-weight: ${props.styles?.tabFontWeight || "normal"};
+    }
+  `;
```

Comment:
Is it possible to use design tokens here instead? https://ant.design/components/tabs#design-token

Reply:
some of the style properties are not available in the design tokens, they can only be changed by overriding antd styles. I have added design token for those that are available. Also, setting global tokens might affect styling in children so I used all the possible component tokens.

Reply:
Got it, let's keep an eye on this then because this might create issues with multiple tab bars.

Reply:
okay